### ISSUE TITLE
import / export

### ISSUE THREAD

**Issue by reggi** [compat-table/node-compat-table#7]
One of the big things for me is support for `import` / `export`, which is in es2015, why isn't this in your list?

**Comment by williamkapke** [comment]
Yeah, it is ES2015 and should be part of the list. Because this project uses [Kangax's tests](https://github.com/kangax/compat-table) it will only display categories they include. Currently, they do not have a test for it. See: https://github.com/kangax/compat-table/issues/316

Node will be using a special file extension to load modules- so that requires any tests to be save to their own files- which will require some reworking of how my scripts run.

The answer for Node (currently) is: Not implemented.

**Comment by ericgrosse** [comment]
What's the point of using ES6 in a node environment if such a key feature is missing?

I did find a polyfill for it [here](https://github.com/ModuleLoader/es6-module-loader), but it would be nice to run a node environment that doesn't require babel in some way, shape or form.

**Comment by ljharb** [comment]
@ericgrosse the loader spec isn't finished, so ES modules don't actually exist yet. `import`/`export` don't actually do anything per the ES6 spec.

**Comment by ghost** [comment]
I'm a little confused. This compat table currently lists ES2015 Support for Node 6.10 and above as 99%. Without [Modules](http://www.ecma-international.org/ecma-262/7.0/index.html#sec-modules) support wouldn't that make the metrics inaccurate?

**Comment by ljharb** [comment]
No, because Modules do not yet exist.

ES6 only contains Module parsing goal semantics (which an engine does not have to implement if they don't provide a way to get there). That's only one piece of "ES Modules".

**Comment by ghost** [comment]
I see. Thanks for clarifying. I believe what I'm looking for is Node support for ES2016, then.

**Comment by ljharb** [comment]
ES2016 over ES2015 only adds `**` and `Array.prototype.includes`. ES2017 adds `async function`, and a number of new API methods. ES Modules, as a whole concept, won't ever be in the ES language spec - just the building blocks for them that ES2015 ships.

**Comment by FossPrime** [comment]
as of 8.7 it's behind a flag.  It's still behind a flag on 9.0

**Comment by jaggli** [comment]
Can we simply add it to the list for good measures? It's also in the [docs](https://nodejs.org/api/esm.html)

**Comment by 0xdevalias** [comment]
See also:

- https://github.com/mdn/content/issues/38335
- https://github.com/mdn/browser-compat-data/issues/26066
- https://github.com/williamkapke/node-compat-table/issues/92
- https://github.com/compat-table/compat-table/issues/316
- https://github.com/compat-table/compat-table/issues/1053
- https://github.com/compat-table/compat-table/issues/1034
- https://github.com/compat-table/compat-table/issues/1580

Tangentially related:

> BCD has compat info for most if not all of the features you describe. See the compat tables on these pages:
> 
> * [developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)
> * [developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import)
> * [developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta)
> * [developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap)
> * [developer.mozilla.org/en-US/docs/Web/API/Worker/Worker](https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker)
> 
> We are working on a mechanism to better group features in collaboration with the [web-platform-dx/feature-set](https://github.com/web-platform-dx/feature-set?rgh-link-date=2023-08-10T08%3A40%3A46.000Z) project. Hopefully, there will be a group for all things modules soon.
> 
> _Originally posted by @Elchi3 in https://github.com/mdn/browser-compat-data/issues/4819#issuecomment-1672805525_
